### PR TITLE
feat(scenario-specification): assign specification candidate

### PR DESCRIPTION
**Pull request by kgajowy**
### Overview

_Please write a description. If the PR is hard to understand, provide a quick explanation of the code._

### Designs

_Link to the related design prototypes (if applicable)_

### Testing instructions

_Please explain how to test the PR: ID of a dataset, steps to reach the feature, etc._

### Feature relevant tickets

_Link to the related task manager tickets_

---

## Checklist before submitting

- [ ] Meaningful commits and code rebased on `develop`.
- [ ] If this PR adds feature that should be tested for regressions when
      deploying to staging/production, please add brief testing instructions
      to the deploy checklist (`docs/deployment-checklist.md`)
- [ ] Update CHANGELOG file

### CODE REVIEW

File: api/apps/api/src/modules/scenario-specification/application/assign-candidate-specification.handler.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  }: AssignCandidateSpecification): Promise<void> {
+    const scenarioSpecification = this.eventPublisher.mergeObjectContext(
+      await this.scenarioSpecificationRepo.findOrCreate(scenarioId),
+    );
+    scenarioSpecification.assignCandidateSpecification(
+      new SpecificationId(specificationId),
+    );
+    await this.scenarioSpecificationRepo.save(scenarioSpecification);
+    scenarioSpecification.commit();
+    return void 0;
```
Comment (Dyostiq) [inline api/apps/api/src/modules/scenario-specification/application/assign-candidate-specification.handler.ts:30]: why? :D

File: api/apps/api/src/modules/scenario-specification/application/assign-candidate-specification.handler.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    specificationId,
+    scenarioId,
+  }: AssignCandidateSpecification): Promise<void> {
+    const scenarioSpecification = this.eventPublisher.mergeObjectContext(
+      await this.scenarioSpecificationRepo.findOrCreate(scenarioId),
+    );
+    scenarioSpecification.assignCandidateSpecification(
+      new SpecificationId(specificationId),
+    );
+    await this.scenarioSpecificationRepo.save(scenarioSpecification);
```
Comment (Dyostiq) [inline api/apps/api/src/modules/scenario-specification/application/assign-candidate-specification.handler.ts:28]: shouldn't it be in a transaction?

File: api/apps/api/src/modules/scenario-specification/application/__tests__/fixtures.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    },
+    ThenCandidateSpecificationChanged: () => {
+      expect(events).toEqual([
+        new CandidateSpecificationChanged(nextCandidateSpecificationId),
+      ]);
+    },
+  };
+};
+
+class InMemoryScenarioSpecificationRepo implements ScenarioSpecificationRepo {
```
Comment (Dyostiq) [inline api/apps/api/src/modules/scenario-specification/application/__tests__/fixtures.ts:74]: I think specification & scenario-specification should be a single module

**Review by Dyostiq — APPROVED** [review]

File: api/apps/api/src/modules/scenario-specification/application/__tests__/fixtures.ts (hunk 3 above)
Comment (kgajowy) [reply api/apps/api/src/modules/scenario-specification/application/__tests__/fixtures.ts:74]: Why?
Why would be the benefits?

File: api/apps/api/src/modules/scenario-specification/application/assign-candidate-specification.handler.ts (hunk 2 above)
Comment (kgajowy) [reply api/apps/api/src/modules/scenario-specification/application/assign-candidate-specification.handler.ts:28]: No need so I believe. Empty scenario specification don't have to be persisted (during `findOrCreate`)

Comment (Dyostiq) [reply api/apps/api/src/modules/scenario-specification/application/assign-candidate-specification.handler.ts:28]: it's find or create, so may lead to lost updates

**Pull request merged**